security/security.c

```diff
@@ -1,2360 +1,6 @@
-			    unsigned long prot)
-{
-	return call_int_hook(file_mprotect, 0, vma, reqprot, prot);
-}
-
-int security_file_lock(struct file *file, unsigned int cmd)
-{
-	return call_int_hook(file_lock, 0, file, cmd);
-}
-
-int security_file_fcntl(struct file *file, unsigned int cmd, unsigned long arg)
-{
-	return call_int_hook(file_fcntl, 0, file, cmd, arg);
-}
-
-void security_file_set_fowner(struct file *file)
-{
-	call_void_hook(file_set_fowner, file);
-}
-
-int security_file_send_sigiotask(struct task_struct *tsk,
-				  struct fown_struct *fown, int sig)
-{
-	return call_int_hook(file_send_sigiotask, 0, tsk, fown, sig);
-}
-
-int security_file_receive(struct file *file)
-{
-	return call_int_hook(file_receive, 0, file);
-}
-
-int security_file_open(struct file *file)
-{
-	int ret;
-
-	ret = call_int_hook(file_open, 0, file);
-	if (ret)
-		return ret;
-
-	return fsnotify_perm(file, MAY_OPEN);
-}
-
-int security_task_alloc(struct task_struct *task, unsigned long clone_flags)
-{
-	int rc = lsm_task_alloc(task);
-
-	if (rc)
-		return rc;
-	rc = call_int_hook(task_alloc, 0, task, clone_flags);
-	if (unlikely(rc))
-		security_task_free(task);
-	return rc;
-}
-
-void security_task_free(struct task_struct *task)
-{
-	call_void_hook(task_free, task);
-
-	kfree(task->security);
-	task->security = NULL;
-}
-
-int security_cred_alloc_blank(struct cred *cred, gfp_t gfp)
-{
-	int rc = lsm_cred_alloc(cred, gfp);
-
-	if (rc)
-		return rc;
-
-	rc = call_int_hook(cred_alloc_blank, 0, cred, gfp);
-	if (unlikely(rc))
-		security_cred_free(cred);
-	return rc;
-}
-
-void security_cred_free(struct cred *cred)
-{
-	/*
-	 * There is a failure case in prepare_creds() that
-	 * may result in a call here with ->security being NULL.
-	 */
-	if (unlikely(cred->security == NULL))
-		return;
-
-	call_void_hook(cred_free, cred);
-
-	kfree(cred->security);
-	cred->security = NULL;
-}
-
-int security_prepare_creds(struct cred *new, const struct cred *old, gfp_t gfp)
-{
-	int rc = lsm_cred_alloc(new, gfp);
-
-	if (rc)
-		return rc;
-
-	rc = call_int_hook(cred_prepare, 0, new, old, gfp);
-	if (unlikely(rc))
-		security_cred_free(new);
-	return rc;
-}
-
-void security_transfer_creds(struct cred *new, const struct cred *old)
-{
-	call_void_hook(cred_transfer, new, old);
-}
-
-void security_cred_getsecid(const struct cred *c, u32 *secid)
-{
-	*secid = 0;
-	call_void_hook(cred_getsecid, c, secid);
-}
-EXPORT_SYMBOL(security_cred_getsecid);
-
-int security_kernel_act_as(struct cred *new, u32 secid)
-{
-	return call_int_hook(kernel_act_as, 0, new, secid);
-}
-
-int security_kernel_create_files_as(struct cred *new, struct inode *inode)
-{
-	return call_int_hook(kernel_create_files_as, 0, new, inode);
-}
-
-int security_kernel_module_request(char *kmod_name)
-{
-	int ret;
-
-	ret = call_int_hook(kernel_module_request, 0, kmod_name);
-	if (ret)
-		return ret;
-	return integrity_kernel_module_request(kmod_name);
+#include <stdio.h>
+void main() {
```

Comment:
`void main()` is bad practice. Use `int main(void)` instead.

Reply:
wth lol hahaha